source/codegen/templates/library.cpp.mako

```diff
@@ -35,10 +43,13 @@ static const char* kLibraryName = "lib${linux_library_name}.so${linux_so_version
 
 namespace ${config["namespace_component"]}_grpc {
 
-${service_class_prefix}Library::${service_class_prefix}Library() : shared_library_(kLibraryName)
+${class_name}::${class_name}() : ${class_name}(std::make_shared<nidevice_grpc::SharedLibrary>()) {}
+
+${class_name}::${class_name}(std::shared_ptr<nidevice_grpc::SharedLibraryInterface> pSharedLibrary) : ${intialization_list}
 {
-  shared_library_.load();
-  bool loaded = shared_library_.is_loaded();
+  p_shared_library_->set_library_name(kLibraryName);
```

Comment:
Why 'p'?

Reply:
Because it's a pointer.
I'm open to changing the name.

Reply:
Does any other pointer variable in this codebase have a "p" prefix? When in Rome, do as the Romans do.

Reply:
Updated the name for the attribute and the constructor parameter.